### ISSUE TITLE
TVPGO

### ISSUE THREAD

**Issue by mtr81** [mbebe/blomqvist#708]
Serwis TVPGO co prawda w necie jeszcze nie ruszył, ale jest dostępny z poziomu aplikacji HBBTV. Na podstawie tej ostatniej napisałem wtyczkę do Kodi umożliwiającej dostęp do kanałów live TVP (wszystkie stacje dostępne w naziemnej TV cyfrowej). Wtyczka nie wymaga logowania. Jest możliwość eksportu listy M3U współpracującej z pluginem. 
[plugin.video.tvpgo.zip](https://github.com/mbebe/blomqvist/files/7988378/plugin.video.tvpgo.zip)

**Comment by hevet** [comment]
@mtr81 super dzięki. Mam pytanie czy we wtyczce do UPC go, którą kiedyś też udostępniłeś, jest możliwość dodania wpisania kodu PIN? We wtyczce od mbebe czasem wyskakuje prośba o PIN i wpisujesz np. 0000 i kanał rusza, a w Twojej wtyczce na tym kanale pojawi się błąd dodatku.

**Comment by zuzia-dev** [comment]
@mtr81 działa, fajnie, że Ci się chciało napisać. @hevet UPC TV Go ma obecnie zaostrzone zasady udostępniania kanałów poza domem wiec jeśli nie jesteś na IP przypisanym do UPC cześć kanałów nie będzie działać (np. TVP). Kiedyś wszystko działo na każdym polskim IP, a teraz niestety już skończyło się.

**Comment by hevet** [comment]
@zuzia-dev nie chodzi mi o kanały TVP. Na niektórych losowo pojawia się prośba o PIN, trzeba go wpisać by kanał zadziałał. Np. Nick Junior itp., a ten działa wszędzie, nawet poza siecią.

Wracając do TVPGO, to na wszystkich lokalnych kanałach TVP3 jest TVP Sport i chyba tak samo jest na TVP World.

**Comment by mtr81** [comment]
Co do TVPGO poniżej poprawka - problem wynikał stąd, że część kanałów (przeważnie tych które były dostępne w TVP Stream) jest w nowym serwisie odmiennie traktowana od pozostałych. Po zmianach zmieniła się struktura linków w liście M3U
[plugin.video.tvpgo.zip](https://github.com/mbebe/blomqvist/files/7992108/plugin.video.tvpgo.zip)

Wtyczce UPC muszę się zatem przyjrzeć, bo oglądam sporadycznie i z w/w błędem się nie spotkałem. Czy w aplikacji UPC bądź wersji przeglądarkowej też coś takiego ma miejsce? I jeszcze a propos tej wtyczki, to czy dalej jest problem z zawieszaniem się strumienia po ok 40-50 minutach oglądania?

**Comment by praspp** [comment]
Tak, w UPC TV Go ten problem występuje nadal i ma związek z autentykacją.

**Comment by hevet** [comment]
@mtr81 postaram się zrobić log jeżeli wyjdzie błąd z PIN-em.

**Comment by hevet** [comment]
@mtr81 dzisiaj wyskoczyła prośba o PIN na BBC Earth
![IMG_20220203_163324](https://user-images.githubusercontent.com/43666290/152375261-69a561e1-c62e-46ec-9753-4ba1f84c78f8.jpg)
Po wpisaniu 0000 kanał się odtwarza.
![IMG_20220203_163340](https://user-images.githubusercontent.com/43666290/152375364-457ac91f-2777-4b57-b2fb-cf32f80af461.jpg)
Natomiast u Ciebie w dodatku wyskakuje błąd UPC i kanał się nie odtwarza.
![IMG_20220203_163446](https://user-images.githubusercontent.com/43666290/152375513-c54cb294-a26b-48ed-9a12-846861c70247.jpg)

**Comment by mtr81** [comment]
@hevet poprawka pluginu UPC umożliwiająca wpisanie PIN-u poniżej.
[plugin.video.upc.zip](https://github.com/mbebe/blomqvist/files/7999753/plugin.video.upc.zip)
Co ciekawe jak wpisywałem inny PIN niż 0000 stream nie ruszał.

**Comment by mtr81** [comment]
Jeśli chodzi o wtyczkę TVPGO, to zmieniłem sposób odtwarzania strumieni (użycie IAS) przez co na wybranych kanałach pojawiła się możliwość cofania trwającego programu do momentu jego rozpoczęcia.
[plugin.video.tvpgo.zip](https://github.com/mbebe/blomqvist/files/7999858/plugin.video.tvpgo.zip)

**Comment by hevet** [comment]
Pewnie dlatego, że domyślny PIN to 0000, można go zmienić logując się na stronie UPC. 
Super, sprawdzę czy działa też u mnie,  dzięki 🙂

**Comment by mtr81** [comment]
Poniżej kolejna wersja wtyczki **TVPGO** uzupełniona o możliwość oglądania programów do 7 dni wstecz (usługa **replay TV**). W związku z przebudową kodu zmieniła się struktura listy M3U. Prosiłbym o przetestowanie czy się poprawnie generuje i kanały  odtwarzają, bo na razie nie mam czasu tego sprawdzić. 
Uwaga - czasami przy przechodzeniu do kolejnych poziomów menu wtyczka może generować błąd. Jest to spowodowane tym, że serwer czasami zwraca błąd (zapytania są realizowane metodą _Persisted Queries Link_). Zazwyczaj po ponowieniu operacji jest ok.
[plugin.video.tvpgo.zip](https://github.com/mbebe/blomqvist/files/8010452/plugin.video.tvpgo.zip)

**Comment by hevet** [comment]
@mtr81 dzisiaj przy próbie wejścia w UPC go wyskakuje błąd. Zdjęcia logu poniżej.
![IMG_20220207_230933](https://user-images.githubusercontent.com/43666290/152881473-253840ef-1381-499f-bac3-499a89bcecb5.jpg)
![IMG_20220207_230950](https://user-images.githubusercontent.com/43666290/152881504-c4d96048-6dea-48f4-bfd6-4dd5a23b3df5.jpg)

**Comment by mtr81** [comment]
Dzięki za info. Dodali dzisiaj kanał MIXTAPE, nie wprowadzając pełnych danych EPG (brak tytułu programu w schemie, czego nie przewidziałem we wtyczce). Poniżej poprawka.
[plugin.video.upc.zip](https://github.com/mbebe/blomqvist/files/8019353/plugin.video.upc.zip)

**Comment by hevet** [comment]
@mtr81 dzięki za szybką pomoc 🙂

**Comment by DenDyGH** [comment]
@mtr81 zapraszamy na nasz kanał discordowy. Chcemy zebrać programistów w jednym miejscu. Chłopaki chętnie pomogą z kwestiami technicznymi, przy okazji ułatwi nam to kontakt z tobą ;)
https://kodipl.net/dev/discord

**Comment by rysson** [comment]
@mtr81 jak się wbijesz na Discord, a mam nadzieję, że będziesz chciał, to daj tutaj zwrotkę, że Ty to Ty.
Np. Swój nick na Discord.

Do zobaczenia!

**Comment by oliwer12345** [comment]
NA JAKIEJ WTYCZCE JEST TO TVPGO

**Comment by hevet** [comment]
Dziwne pytanie...
Tutaj w temacie masz https://github.com/mbebe/blomqvist/issues/708#issuecomment-1030851573

**Comment by oliwer12345** [comment]
DZIEKI

**Comment by mtr81** [comment]
Poniżej poprawki kodu do **TVPGO**, po których wtyczka działa stabilniej. Poza tym w kategorii replay nie pokazują się już programy (audycje), które się jeszcze nie rozpoczęły
[plugin.video.tvpgo.zip](https://github.com/mbebe/blomqvist/files/8037756/plugin.video.tvpgo.zip)

Ze spraw organizacyjnych na Discordzie mam ten sam nick co tutaj. Dzięki za zaproszenie do grupy :)

**Comment by xxcriticxx** [comment]
Czy dziala tvpgo przez vpn?
